src/genome.py

```diff
@@ -214,17 +341,40 @@ def disable_te(self, te: int) -> None:
         TEs are already inactive, so there is no need to do anything
         for those.
         """
-        ...  # FIXME
+        # update dictionary
+        self.TEs[te][2] = 'D'
+        # update genome
+        pos = self.TEs[te][0]
+        length = self.TEs[te][1]
+
+        count = 0
+
+        current = self.head.next
+        # iterator
+        while current is not self.head:
+            if count == pos:
+                for _ in range(length):
+                    # change link value to 'x'
+                    current.val = 'x'
+                    current = current.next
+                break
+            current = current.next
```

Comment:
If you had a `dict`, mapping each TE to its first link in the linked list you wouldn't have to do a linear search here.

Reply:
How do I do that without indexing?